dvc/fs/gdrive.py

```diff
@@ -208,15 +209,25 @@ def _drive(self):
         from pydrive2.auth import GoogleAuth
         from pydrive2.drive import GoogleDrive
 
-        if os.getenv(GDriveFileSystem.GDRIVE_CREDENTIALS_DATA):
-            with open(self._gdrive_user_credentials_path, "w") as cred_file:
+        temporary_save_path = self._gdrive_user_credentials_path
+        is_credentials_temp = os.getenv(
+            GDriveFileSystem.GDRIVE_CREDENTIALS_DATA
+        )
+        if self._use_service_account:
```

Comment:
`GDRIVE_CREDENTIALS_DATA` this way has two difference meanings in different situations. I think this is confusing, complicates the code, etc. Can we just do the same as we had before? If we don't change anything and just revert the CI variable value, I think it should work? Or cache the saved credentials instead of the raw JSON file?

Reply:
I found that logic a bit weird, considering it would throw a random error when somebody tries to run `GDRIVE_CREDENTIALS_DATA=... dvc status -c` in case of `...` is the raw JSON file.

Reply:
yep, I suggest we do a separate env variable - GDRIVE_SERVICE_ACCOUNT_DARA?  It's bad to mix two different meanings into the same variable.

> considering it would throw a random error when somebody tries to run

handle and show a nice message?

Reply:
> handle and show a nice message?

It is an arbitrary key error :( Probably won't even worth the effort though if we want, I guess the best place is to that is wrapping around it in the pydrive2.

Reply:
Agreed. 

```
 Raises:
            ValueError, if the credential type is not :data:`SERVICE_ACCOUNT`.
            KeyError, if one of the expected keys is not present in
                the keyfile.
```

handle this in the `auth.py` `ServiceAuth()` (?) makes total sense